iot_config/src/gateway_service.rs

```diff
@@ -17,48 +15,49 @@ use node_follower::{
     follower_service::FollowerService,
     gateway_resp::{GatewayInfo, GatewayInfoResolver},
 };
-use sqlx::{Pool, Postgres};
 use tonic::{Request, Response, Status};
 
 pub struct GatewayService {
-    admin_pubkey: PublicKey,
     follower_service: FollowerService,
-    pool: Pool<Postgres>,
     region_map: RegionMap,
     signing_key: Keypair,
 }
 
 impl GatewayService {
-    pub async fn new(settings: &Settings) -> Result<Self> {
-        let pool = settings.database.connect(10).await?;
-        let region_map = RegionMap::new(&pool).await?;
+    pub fn new(settings: &Settings, region_map: RegionMap) -> Result<Self> {
         Ok(Self {
-            admin_pubkey: settings.admin_pubkey()?,
             follower_service: FollowerService::from_settings(&settings.follower),
-            pool,
             region_map,
             signing_key: settings.signing_keypair()?,
         })
     }
-
-    fn verify_admin_signature<R>(&self, request: R) -> Result<R, Status>
-    where
-        R: MsgVerify,
-    {
-        request
-            .verify(&self.admin_pubkey)
-            .map_err(|_| Status::permission_denied("invalid admin signature"))?;
-        Ok(request)
-    }
 }
 
 #[tonic::async_trait]
 impl iot_config::Gateway for GatewayService {
     async fn location(
         &self,
-        _request: Request<GatewayLocationReqV1>,
+        request: Request<GatewayLocationReqV1>,
     ) -> GrpcResult<GatewayLocationResV1> {
-        unimplemented!()
+        // Should this rpc be admin-authorized only or should a requesting pubkey
```

Comment:
My understanding right now is only HPR's will be requesting gateway locations.

Reply:
@madninja and i are discussing whether or not this needs to become a more generic `metadata` method for retrieving any on-chain data about a hotspot, inclusive of the asserted location but that would be either primarily or exclusively for the other oracles so between that and the HPRs still under the umbrella of "network infrastructure" so likely authenticated but we don't currently authenticate any requests the oracles make (they only request from their host-local blockchain-node instance) so request signing would be a new requirement for them